### PR TITLE
Automatic parsing of terms only finds terms that are prefixed with given string

### PR DESCRIPTION
**Pull request by panpomaly**
Hello,

i'm currently working on a project, where it is not an option to automatically replace all occurrences of all terms in the glossary, but it is still necessary to automatically replace them. The problem is, that depending on context the term has a different meaning and not all should be replaced.

Example (german): 
ab - is a german word which can occur everywhere in text, but should not be replaces
ab - is a abbreviation in the glossary with a completely different meaning

The customer already delivered the content and prefixed the terms she wanted to replace with an Asterisk. So i just extended dpn_glossary to only replace terms that are prefixed.

* Configurable with {$plugin.tx_dpnglossary.settings.termPrefix}
* Default is the empty string, which won’t change the behaviour of ParserService.php
* If a string is given, the term has to be prefixed by it that the automatic parser exchange wraps it

I hope it can be merged, so we don't have to make a fork of dpn_glossary for the project.

Thanks,
Stephan

### CODE REVIEW

**Comment by featdd** [comment]
Hm this is a stupid problem I see...
But having all terms to be prefixed seems as a huge bunch of work to me.
Maybe adding a field to a term (a checkbox) that says the parser it has to be prefixed.

But before merging I would take a bit time and think about alternative and maybe more efficient solutions :-)

File: Classes/Service/ParserService.php [hunk 1]
```diff
@@ -362,7 +362,7 @@ protected function regexParser(&$text, Term $term, &$replacements, $wrappingCall
          */
         $regex = self::REGEX_DELIMITER .
             '(^|\G|[\s\>[:punct:]]|\<br*\>)' .
-            '(' . preg_quote($term->getName(), self::REGEX_DELIMITER) . ')' .
+            preg_quote($this->settings['termPrefix']) . '(' . preg_quote($term->getName(), self::REGEX_DELIMITER) . ')' .
```
Comment (featdd) [inline Classes/Service/ParserService.php]: Missing the delimiter as second parameter
see e87eb34a82440405dc5f80fe44ec1e324717b187 why

**Comment by panpomaly** [comment]
Thanks for your fast reply!

In our project the prefixing is no problem, but i can see that it is not ideal for a generic solution. The checkbox on every term seems like a good idea.

**Comment by featdd** [comment]
Hay,
You had any helpful ideas yet? I think if we do this there also has to be a rte integration.
I will do some investigation how to add a button and make the prefixing easier for the integrator, or maybe you have the knowhow to amend this to your pull request.

Sorry for the late reply, had to work a lot :-D

**Comment by featdd** [comment]
@panpomaly I added a feature branch, would be so kind to merge Into it?
I'm on the rte plugin I mentioned above :-)

**Comment by panpomaly** [comment]
Hi featdd,

i also had to work a lot and mostly on other projects, but i set a reminder to lock into this again for today. You already invested your time to add the settings to terms and synonyms. Thank you!

My knowledge about extending the rte is not existing and i am not sure it is needed, because it just adds the the prefix, but may be you got something fancier on your mind ? 

I am not sure i will get to do something (work on or test) the new code/branch this week, but i should find some free time in the 2 weeks after that (holidays, yeah!). 

Thanks!

**Comment by featdd** [comment]
Instead of prefixing I'm thinking of let this "button" in RTE lets you wrap a word in a span with a special class or something.

Prefixing is not only heavy work, also it can be visible in frontend if the parser has an excluded page or any other failsafe etc.

So I think the better solution will be "span" and to make that easy I want to add a special RTE button so it's just a select and a klick to define the word explicit parsable

PS: lets move conversation to #76

**Pull request merged**